### PR TITLE
Add system logs store, ref. DCOS-9722

### PR DESCRIPTION
**Pull request by mlunoe**
---
~~⚠️ Depends on the Actions file from this PR: https://github.com/dcos/dcos-ui/pull/1331~~ Merged!

---

This PR introduces a store for the system logs.

You can test the action and store out here (Not complete implementation, but gives you the ability to play around with it): https://github.com/dcos/dcos-ui/tree/mlunoe/feature/add-new-stream-logs-view.

You need to start a cluster with `testing/pull/977` to test it out.

(Sorry, for the size, most of it is tests though)

### CODE REVIEW

**Comment by mlunoe** [comment]
**mesosphere-ci** Retest this please

File: src/js/stores/SystemLogStore.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import {
+  SYSTEM_LOG_CHANGE,
+  SYSTEM_LOG_REQUEST_ERROR
+} from '../constants/EventTypes';
+import BaseStore from './BaseStore';
+import SystemLogActions from '../events/SystemLogActions';
+import Util from '../utils/Util';
+
+// Max data storage
+const MAX_FILE_SIZE = 250000000;
```
Comment (mlunoe) [inline src/js/stores/SystemLogStore.js]: Yes! Added a description, and updated the value to something a little bit more describing

File: src/js/stores/SystemLogStore.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      ) || 0;
+    }, 0);
+
+    // Update new size
+    newLogData.totalSize += length;
+    // Remove entires until we have room for next entry
+    while (newLogData.totalSize > 0 && newLogData.entries.length > 0 &&
+      newLogData.totalSize > MAX_FILE_SIZE) {
+      let removedEntry;
+      if (eventType === 'append') {
```
Comment (mlunoe) [inline src/js/stores/SystemLogStore.js]: done!

File: src/js/stores/SystemLogStore.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+  processLogEntry(subscriptionID, entry = {}) {
+    if (!this.logs[subscriptionID]) {
+      this.logs[subscriptionID] = {entries: [], totalSize: 0};
+    }
+
+    this.logs[subscriptionID] = this.addEntries(
+      this.logs[subscriptionID],
+      [entry],
+      'append'
```
Comment (mlunoe) [inline src/js/stores/SystemLogStore.js]: done!

File: src/js/stores/SystemLogStore.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    if (!this.logs[subscriptionID]) {
+      this.logs[subscriptionID] = {entries: [], totalSize: 0};
+    }
+
+    this.logs[subscriptionID].hasLoadedTop = firstEntry;
+
+    this.logs[subscriptionID] = this.addEntries(
+      this.logs[subscriptionID],
+      entries,
+      'prepend'
```
Comment (mlunoe) [inline src/js/stores/SystemLogStore.js]: Done!

File: src/js/stores/SystemLogStore.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    );
+
+    this.emit(SYSTEM_LOG_CHANGE, subscriptionID, 'prepend');
```
Comment (mlunoe) [inline src/js/stores/SystemLogStore.js]: Done!

**Comment by mlunoe** [comment]
@nLight All comments addressed!

**Comment by nLight** [comment]
Awesome work! 🎉

**Comment by Poltergeist** [comment]
LGTM, Lets merge this.

**Pull request merged**